ostree_upload_server/flatpak_importer.py

```diff
@@ -134,15 +252,34 @@ def import_flatpak(flatpak,
 
         # See if the ref is already pointed at this commit
         _, current_rev = repo.resolve_rev(metadata['ref'], allow_noent=True)
+        logging.debug('Current {} commit: {}'.format(metadata['ref'],
+                                                     current_rev))
         if current_rev == commit:
             logging.info('Ref {} already at commit {}'
                          .format(metadata['ref'], commit))
             return
+        elif current_rev is not None:
+            # See if the current ref was built from this commit.
+            # Normally you'd read each commit and have ostree check if
+            # the roots are equal, but we don't have the new commit
+            # until the delta is executed. Since that's slow, fast path
+            # this by checking whether the tree checksums from the
+            # commits match like ostree_repo_file_equal() eventually
+            # does.
+            commit_variant = delta.get_child_value(4)
+            _, current_variant, _ = repo.load_commit(current_rev)
+
+            # The tree content checksum is at index 6 while the tree
+            # metadata checksum is at index 7.
+            if (commit_variant[6] == current_variant[6]) and \
+               (commit_variant[7] == current_variant[7]):
```

Comment:
Would be good to make these numbers turned into constants for easier maintenance since it wouldn't require comments for them either.

Reply:
OK